### PR TITLE
MM-1409 Adding ability to export data from mattermost

### PR DESCRIPTION
**Pull request by crspeller**
## This is a preview feature
- Format of exported data is not final. This is to be used for preview / backup purposes only. Do not expect to be able to import this into future versions of Mattermost (yet). 

### TODO:
- Unit tests will be completed with the ability to import what is exported in this commit.
- Some of the code is present but not functional. It is in anticipation of the improvements to this feature that will come with the admin console. 
- Chunking and other improvements so this won't fail with large exports. 
- Integration with upcoming Admin console.
- Selection of which teams/channels/users should be exported. 
- Integration with upcoming job scheduler.

### CODE REVIEW

File: store/sql_channel_store.go [hunk 1]
```diff
@@ -678,3 +678,25 @@ func (s SqlChannelStore) UpdateNotifyLevel(channelId, userId, notifyLevel string
...
 }
+
+func (s SqlChannelStore) GetForExport(teamId string) StoreChannel {
+	storeChannel := make(StoreChannel)
+
+	go func() {
+		result := StoreResult{}
+
+		var data []*model.Channel
+		_, err := s.GetReplica().Select(&data, "SELECT * FROM Channels WHERE TeamId = :TeamId AND DeleteAt = 0 AND Type = 'O'", map[string]interface{}{"TeamId": teamId})
```
Comment (coreyhulen) [inline store/sql_channel_store.go]: Probably more a spec issue, but why not allow export of deleted channels.  Nothing to fix, just a question for PMs.

**Comment by coreyhulen** [comment]
+1

File: store/sql_channel_store.go (hunk 1 above)
Comment (crspeller) [inline store/sql_channel_store.go]: If PM wants this, this should be an option.

**Comment by jwilander** [comment]
+1

**Comment by gabel** [comment]
:+1: +1

**Pull request merged**